subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy

```diff
@@ -175,6 +175,38 @@ abstract class AbstractSmokeTest extends Specification {
         settingsFile = new File(testProjectDir.root, "settings.gradle")
     }
 
+    def withPluginValidation() {
+        def isKotlin = defaultBuildFileName.endsWith('.gradle.kts')
+        def pluginsBlockIdx = buildFile.text.indexOf('plugins {')
+        def buildScriptBlockIdx = buildFile.text.indexOf('buildscript {')
+        if (pluginsBlockIdx > 0 || (pluginsBlockIdx == -1 && buildScriptBlockIdx == -1)) {
+            buildFile.text = buildFile.text.replace('plugins {', """
+            plugins {
+                ${isKotlin ? '`validate-external-gradle-plugin`' : 'id "validate-external-gradle-plugin"'}
+        """)
+        } else if (buildScriptBlockIdx > 0) {
+            throw new UnsupportedOperationException("Validation with buildscript block isn't supported in smoke tests yet")
+        }
+    }
+
+    BuildResult checkThatPluginValidationFailsWhen(Closure<Boolean> shouldFail) {
+        def failsValidation =  shouldFail()
+        def validation = runner('validateExternalPlugins')
+        if (failsValidation) {
+            validation.buildAndFail()
+        } else {
+            validation.build()
+        }
+    }
```

Comment:
I think it would be important to check for exact validation errors by text, not just whether or not the validation failed. Essentially the same as we do with deprecation warnings. Doing so would allow us to a) document which validations are broken for a given version of the plugin, and b) still fail the build if newly added validations break the plugin more.